finance_research/parse/tabular_parser.py

```
@@ -0,0 +1,78 @@
+def parse_csv_bytes(
+    data: bytes, encoding: str = "utf-8"
+) -> Tuple[pd.DataFrame, str]:
+    """Parse CSV bytes into a single DataFrame.
+
+    Tries the requested encoding first, then a list of common fallbacks
+    (utf-8-sig for BOM, gb18030 for Chinese Excel exports).
+    """
+    for enc in (encoding, "utf-8-sig", "gb18030", "latin-1"):
+        try:
+            df = pd.read_csv(io.BytesIO(data), encoding=enc)
+            return df, f"CSV 表格（编码 {enc}）"
```

Ruff (0.16.1)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)